Tella/Domain/Entity/Report/Vault/VaultFileDB.swift

```diff
@@ -0,0 +1,293 @@
+        height = try container.decode(Double.self, forKey: .height)
+        
+        
+    }
+    
+    init(dictionnary: [String:Any])   {
+        
+        let id = dictionnary[CodingKeys.id.rawValue] as? String
+        
+        let typeInt = dictionnary[CodingKeys.type.rawValue] as? Int
+        let type = VaultFileType(rawValue: typeInt ?? 0) ?? .unknown
+        
+        
+        let thumbnail = dictionnary[CodingKeys.thumbnail.rawValue] as? Data
+        let name = dictionnary[CodingKeys.name.rawValue] as? String
+        
+        
+        let createdDouble = dictionnary[CodingKeys.created.rawValue] as? Double
+        let created = createdDouble?.getDate() ?? Date()
+        
+        
+        let duration = dictionnary[CodingKeys.duration.rawValue] as? Double
+        
+        
+        let size = dictionnary[CodingKeys.size.rawValue] as? Int
+        let mimeType = dictionnary[CodingKeys.mimeType.rawValue] as? String
+        
+        let width = dictionnary[CodingKeys.width.rawValue] as? Double
+        let height = dictionnary[CodingKeys.height.rawValue] as? Double
+        
+        
+        
+        
+        self.id = id ?? ""
+        self.type = type
+        self.thumbnail = thumbnail
+        self.name = name ?? ""
+        self.created = created
+        self.duration = duration
+        self.size = size ?? 0
+        self.mimeType = mimeType
+        self.width = width
+        self.height = height
+        
+        
+    }
+}
+
+extension VaultFileDB: CustomDebugStringConvertible {
+    
+    var debugDescription: String {
+        return "\(type): \(String(describing: name)), \(id)"
+    }
+    
+}
+
+
+extension VaultFileDB {
+    
+    var thumbnailImage: UIImage {
+        
+        guard let thumbnail else {return UIImage()}
+        
+        guard let image = UIImage(data: thumbnail) else {return UIImage()}
+        
+        return image
+    }
+    
+    
+    
+    var iconImage: UIImage {
+        
+        switch type {
+            
+        case .directory:
+            return #imageLiteral(resourceName: "filetype.small_folder")
+            
+        case .file:
+            switch mimeType?.tellaFileType {
+            case .audio:
+                return #imageLiteral(resourceName: "filetype.small_audio")
+            case .document:
+                return #imageLiteral(resourceName: "filetype.small_document")
+            case .video:
+                return #imageLiteral(resourceName: "filetype.small_video")
+            case .image:
+                return UIImage()
+            case .other:
+                return #imageLiteral(resourceName: "filetype.small_document")
+                
+            default:
+                return #imageLiteral(resourceName: "filetype.small_document")
+            }
+            
+        case .unknown:
+            return #imageLiteral(resourceName: "filetype.small_document")
+            
+        }
+    }
+    
+    var bigIconImage: UIImage {
```

Comment:
for this type of `complex` operations i think we could use a method instead of a property. Also I think we can re use some logic in `iconImage` and `bigIconImage`